### ISSUE TITLE
Takeoff command blows up

### ISSUE THREAD

**Issue by miklosduma** [ProfHoekstra/bluesky#108]
Hi Guys,

I've tried running:
```
SP01 ADDWPT EGKK/RW08R
SP01 ADDWPT TAKEOFF
```

And received the following after `TAKEOFF`:
```
File "bluesky/traffic/route.py", line 222, in addwptStack
    wpidx = self.addwpt(idx, name, wptype, lat, lon, alt, spd, afterwp, beforewp)
UnboundLocalError: local variable 'beforewp' referenced before assignment
```

Running only:
```
SP01 ADDWPT TAKEOFF
```
Returns:
```
Traceback (most recent call last):
  File "BlueSky_qtgl.py", line 113, in <module>
    main_loop()
  File "BlueSky_qtgl.py", line 103, in main_loop
    manager.run()
  File "../bluesky/bluesky/simulation/qtgl/nodemanager.py", line 24, in run
    bs.sim.doWork()
  File "../bluesky/bluesky/simulation/qtgl/simulation.py", line 98, in doWork
    stack.process()
  File "/../bluesky/bluesky/stack/stack.py", line 1158, in process
    results = function(*parser.arglist)  # * = unpack list to call arguments
  File "/../bluesky/bluesky/stack/stack.py", line 149, in <lambda>
    lambda idx, *args: bs.traf.ap.route[idx].addwptStack(idx, *args),
  File "/../bluesky/bluesky/traffic/route.py", line 213, in addwptStack
    elif self.wptype[0]==self.orig:
IndexError: list index out of range
```

I may be using the commands incorrectly but in other cases you send helpful error messages and the simulation does not blow up.

Cheers,
Miklos

p.s.: Tested with Python 2.7 and Python 3.6

**Comment by jooste** [comment]
Hi Miklos,

There were indeed two bugs. They should be fixed in e84c25b. I tested this as follows (slightly different than what you describe, but I assume you did something similar, since you need a CRE command as well):

CRE SP01 B744 EGKK/RW08R * 0 0
SP01 ADDWPT TAKEOFF

With the above-referenced commit this now works properly.

**Comment by ProfHoekstra** [comment]
Ook testfile in testscenarios geprobeerd? Die bevat commando’s die hij noemt. - Jacco
